tests/Oryx.Integration.Tests/SqlServerIntegrationTests.cs

```diff
@@ -3,50 +3,176 @@
-    //        await RunTestAsync("python", "3.7", Path.Combine(HostSamplesDir, "python", "mssqlserver-sample"));
-    //    }
-
-    //    [Theory]
-    //    [InlineData("7.3")]
-    //    [InlineData("7.2")]
-    //    // pdo_sqlsrv only supports PHP >= 7.1
-    //    public async Task PhpApp_UsingPdo(string phpVersion)
-    //    {
-    //        await RunTestAsync(
-    //            "php",
-    //            phpVersion,
-    //            Path.Combine(HostSamplesDir, "php", "sqlsrv-example"),
-    //            8080,
-    //            specifyBindPortFlag: false);
-    //    }
-    //}
+    [Trait("category", "db")]
+    [Trait("db", "sqlserver")]
+    public class SqlServerIntegrationTests : PlatformEndToEndTestsBase
+    {
+        private const string DbServerHostnameEnvVarName = "SQLSERVER_DATABASE_HOST";
+        private const string DbServerUsernameEnvVarName = "SQLSERVER_DATABASE_USERNAME";
+        private const string DbServerPasswordEnvVarName = "SQLSERVER_DATABASE_PASSWORD";
+        private const string DbServerDatabaseEnvVarName = "SQLSERVER_DATABASE_NAME";
+
+        private const int ContainerPort = 3000;
+        private const string DefaultStartupFilePath = "./run.sh";
+
+        private readonly ITestOutputHelper _output;
+        private readonly string _hostSamplesDir;
+        private readonly string _tempRootDir;
+
+        public SqlServerIntegrationTests(ITestOutputHelper output, TestTempDirTestFixture testTempDirTestFixture)
+        {
+            _output = output;
+            _hostSamplesDir = Path.Combine(Directory.GetCurrentDirectory(), "SampleApps");
+            _tempRootDir = testTempDirTestFixture.RootDirPath;
+        }
+
+        [Fact]
+        public async Task NodeApp_MicrosoftSqlServerDB()
+        {
+            // Arrange
+            var appName = "node-mssql";
+            var hostDir = Path.Combine(_hostSamplesDir, "nodejs", appName);
+            var volume = DockerVolume.CreateMirror(hostDir);
+            var appDir = volume.ContainerDir;
+            var script = new ShellScriptBuilder()
+                .AddCommand($"oryx -appPath {appDir} -bindPort {ContainerPort}")
+                .AddCommand(DefaultStartupFilePath)
+                .ToString();
+
+            await EndToEndTestHelper.BuildRunAndAssertAppAsync(
+                appName,
+                _output,
+                new List<DockerVolume> { volume },
+                Settings.BuildImageName,
+                "oryx",
+                new[] { "build", appDir, "-l", "nodejs", "--language-version", "10.14" },
+                "oryxdevms/node-10.14",
+                GetEnvironmentVariables(),
+                ContainerPort,
+                "/bin/bash",
+                new[]
+                {
+                    "-c",
+                    script
+                },
+                async (hostPort) =>
+                {
+                    var data = await _httpClient.GetStringAsync($"http://localhost:{hostPort}/");
```

Comment:
One hiccup that I ran into was firewalls on Azure do not allow requests from random IP addresses to connect to the database server. For now, I gave a range of IP addresses which essentially is like having no firewall, but ideally we should have a restricted range.